context/app/static/js/components/Detail/TableOfContents.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import Typography from '@material-ui/core/Typography';
+import List from '@material-ui/core/List';
+import Link from '@material-ui/core/Link';
+import { throttle } from '../../helpers/functions';
+
+const TableContainer = styled.div`
+  margin-right: 75px;
+`;
+
+const StickyNav = styled.nav`
+  position: sticky;
+  top 70px;
+`;
+
+const TableTitle = styled(Typography)`
+  margin-left: 7px;
+`;
+
+const StyledItemLink = styled(Link)`
+  font-size: .8125rem;
+  line-height: 2;
+  padding-left: 4px;
+  border-left: 3px solid transparent;
+  &:hover {
+    border-left: 3px solid #C4C4C4;
+  }
+  ${(props) => props.$isCurrentSection
+    && css`
+    color: #3781D1;
+    border-left: 3px solid #C4C4C4;
+  `};
+`;
+
+function ItemLink(props) {
+  const {
+    item, currentSection, handleClick,
+  } = props;
+  return (
+    <StyledItemLink
+      display="block"
+      color={currentSection === item.hash ? 'textPrimary' : 'textSecondary'}
+      href={`#${item.hash}`}
+      underline="none"
+      onClick={handleClick(item.hash)}
+      $isCurrentSection={currentSection === item.hash}
+    >
+      {item.text}
+    </StyledItemLink>
+  );
+}
+
+function getItemsClient(headings) {
+  const itemsWithNode = [];
+
+  headings.forEach((item) => {
+    itemsWithNode.push({
+      ...item,
+      node: document.getElementById(item.hash),
+    });
+  });
+  return itemsWithNode;
+}
+
+function useThrottledOnScroll(callback, delay) {
```

Comment:
There's a lot of complexity here. At first look, I would prefer just to use the browser's built-in handling for anchor links... but if that's not good enough, add a comment explaining why. Thanks!

Reply:
Nils wanted this functionality specifically.